mcx/sample.py

```diff
@@ -385,9 +385,13 @@ def sample_scan(
     """
     num_samples = rng_keys.shape[0]
 
-    progress_bar_scan, set_tqdm_description, close_tqdm = progress_bar_factory(
-        num_samples
+    progress_bar = tqdm(range(num_samples))
+    progress_bar.set_description(
+        f"Collecting {num_samples:,} samples across {num_chains:,} chains",
+        refresh=False,
     )
+    print_rate = int(num_samples / 10)
+    progress_bar_scan = progress_bar_factory(progress_bar, print_rate)
```

Comment:
I moved the choice of `print_rate` to `sample_scan` and pass it it to `progress_bar_factory`. 
The function `_updated_progress_bar` in `jax.py` can then be simplified a bit as you don't need to pass in `print_rate`

Reply:
Great idea!